connection_test.go

```diff
@@ -1079,3 +1084,104 @@ func TestContextCanceledOnTCPClose(t *testing.T) {
 		clientCh.Close()
 	})
 }
+
+// getConnection returns the introspection result for the unique inbound or
+// outbound connection. An assert will be raised if there is more than one
+// connection of the given type.
+func getConnection(t *testing.T, ch *Channel, direction int) *ConnectionRuntimeState {
+	state := ch.IntrospectState(&IntrospectionOptions{
+		IncludeEmptyPeers: false,
+	})
+
+	for _, peer := range state.RootPeers {
+		var connections []ConnectionRuntimeState
+		if direction == inbound {
+			connections = peer.InboundConnections
+		} else {
+			connections = peer.OutboundConnections
+		}
+
+		assert.True(t, len(connections) <= 1, "Too many connections found")
+		if len(connections) == 1 {
+			return &connections[0]
+		}
+	}
+
+	assert.Fail(t, "No connections found")
+	return nil
```

Comment:
maybe make this `panic("not reachable")`

Reply:
It is reachable, if the test has a problem and no connections were established.

Reply:
is it? shouldn't `assert.Fail` stop execution straight away by killing the current goroutine?

Reply:
Changed it to FailNow()